src/wp-includes/global-styles-and-settings.php

```diff
@@ -306,8 +306,30 @@ function wp_add_global_styles_for_blocks() {
 
 	$tree        = WP_Theme_JSON_Resolver::get_merged_data();
 	$block_nodes = $tree->get_styles_block_nodes();
+
+	$can_use_cached = ! wp_is_development_mode( 'theme' );
+	if ( $can_use_cached ) {
+		$settings_hash = _wp_get_global_settings_hash();
```

Comment:
Since we're not reusing this hash anywhere else, we can probably keep all of the hash generation logic here, rather creating a separate private function. It would also be helpful to add an inline comment explaining why `wp_get_global_settings()` needs to be used for cache busting.